kasa/smart/smartmodule.py

```diff
@@ -179,12 +188,20 @@ def data(self) -> dict[str, Any]:
 
         filtered_data = {k: v for k, v in dev._last_update.items() if k in q_keys}
 
+        remove_keys: list[str] = []
         for data_item in filtered_data:
             if isinstance(filtered_data[data_item], SmartErrorCode):
-                raise DeviceError(
-                    f"{data_item} for {self.name}", error_code=filtered_data[data_item]
-                )
-        if len(filtered_data) == 1:
+                if data_item in self.optional_response_keys:
+                    remove_keys.append(data_item)
```

Comment:
Any downsides if we'd just keep the error code available, just in case if it could be useful for some reason?

Reply:
I think it's better to remove the key so that modules are not constantly testing for errors and can instead use `get("blah", {})` for the optional keys. Instead I've added logging that will log once when it's removing a key and stop logging after that to avoid spamming.